### PR TITLE
functors task (1) - Panasenko

### PR DESCRIPTION
**Pull request by pro100Daim**
MPanasenko 
functors task (1)

### CODE REVIEW

**Comment by smay1613** [comment]
Please fix pull request title

File: functors task (1)/student/include/impl.h [hunk 1]
```diff
@@ -8,6 +8,21 @@
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()() {
+
+        static BeerBrand beer = BeerBrand::Leffe;
+
+        switch(beer){
+        case BeerBrand::HoeGaarden: beer = BeerBrand::Corona; break;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:16]: and what if we have 20+ beer brands? :)

File: functors task (1)/student/include/impl.h [hunk 2]
```diff
@@ -40,9 +57,27 @@ struct MixingPolicy
...
+            } else if(NonAlcoholDrink::GrapefruitJuice == drink_nalc) {
+                result = Cocktail::Greyhount;
+            }
+            break;
+        case AlcoholDrink::Whiskey:
+            if(NonAlcoholDrink::SevenUp == drink_nalc){
+                result = Cocktail::SevenPlusSeven;
+            }
+            break;
+        default:
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:76]: why do you need this empty "default"?
you can return "Oops" here

**Review by smay1613 — CHANGES_REQUESTED** [review]

File: functors task (1)/student/include/impl.h [hunk 3]
```diff
@@ -10,18 +10,14 @@ struct BeerOrganizer
...
-        case BeerBrand::HoeGaarden: beer = BeerBrand::Corona; break;
-        case BeerBrand::Corona:     beer = BeerBrand::Carlsberg; break;
-        case BeerBrand::Carlsberg:  beer = BeerBrand::Bud; break;
-        case BeerBrand::Bud:        beer = BeerBrand::ZlataPraha; break;
-        case BeerBrand::ZlataPraha: beer = BeerBrand::Leffe; break;
-        case BeerBrand::Leffe:      beer = BeerBrand::HoeGaarden; break;
-        default: break;
+        if(static_cast<int>(BeerBrand::Max) == (curr_beer + 1))
+        {
+            curr_beer = 0;
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:17]: magic number

File: functors task (1)/student/include/impl.h [hunk 4]
```diff
@@ -10,18 +10,14 @@ struct BeerOrganizer
 {
     BeerBrand operator()() {
 
-        static BeerBrand beer = BeerBrand::Leffe;
+        static int curr_beer {0};
```
Comment (smay1613) [inline functors task (1)/student/include/impl.h:13]: don't use abbreviations

File: functors task (1)/student/include/impl.h (hunk 1 above)
Comment (pro100Daim) [reply functors task (1)/student/include/impl.h]: Thanks for reply!
Switch is removed.

**Pull request closed without merging**